middleware/kubernetes/kubernetes.go

```diff
@@ -227,40 +269,42 @@ func (k *Kubernetes) Records(name string, exact bool) ([]msg.Service, error) {
 	return records, nil
 }
 
-func (k *Kubernetes) getRecordsForServiceItems(serviceItems []*api.Service, zone string) []msg.Service {
+func endpointHostname(addr api.EndpointAddress) string {
+	if addr.Hostname != "" {
+		return strings.ToLower(addr.Hostname)
+	} else {
+		if strings.Contains(addr.IP, ".") {
+			return strings.Replace(addr.IP, ".", "-", -1)
+		}
+		if strings.Contains(addr.IP, ":") {
```

Comment:
addr.IP is the string IP exactly as returned to us from the kubernetes API.  So, we are trustting the data we get back from kubernetes to be correct.  Its not necessarily important that this value actually be a valid IP.  What matters is that the addr.Hostname (or if blank, the addr.IP) uniquely identifies the endpoint in the context of the service/namespace.  IPv6 collapsing doesnt jeopardize that.

This function returns